lib/MongoDB/_Server.pm

```diff
@@ -82,6 +83,24 @@ has is_master => (
     default => sub { {} },
 );
 
+has compressor => (
+    is => 'lazy',
+    isa => Maybe[Str],
+    builder => '_build_compressor',
+);
+
+has zlib_compression_level => (
+    is => 'ro',
+    isa => ZlibCompressionLevel,
+    default => sub { -1 },
+);
+
+sub _build_compressor {
+    my ($self) = @_;
+    my ($comp) = @{ ($self->is_master || {})->{compression} || [] };
```

Comment:
A subtlety here: I don't think the server is required to return compressors in the preference order sent by the client, only an intersection.  The client could send "zlib, snappy" and the server could reply with "snappy, zlib".  So server probably needs to get a "preferred_compressors" attribute and this should take the servers reply in preference order.